crates/web-plugins/didcomm-messaging/protocols/discover-features/src/handler.rs

```diff
@@ -0,0 +1,302 @@
+                        .get("id")
+                        .unwrap()
+                        .to_string(),
+                )
+                .unwrap();
+
+                assert_eq!(id, MEDIATION.to_string());
+            }
+            Err(e) => {
+                panic!("This should not occur {:?}", e)
+            }
+        }
+    }
+    #[tokio::test]
+    async fn test_get_supported_protocols_with_wildcard() {
+        let queries = json!({"feature-type": "protocol", "match": "https://didcomm.org/coordinate-mediation/2.*"});
+
+        let body = Queries {
+            queries: vec![queries],
+        };
+        let id = Uuid::new_v4().urn().to_string();
+
+        let message = Message::build(id, QUERY_FEATURE.to_string(), json!(body)).finalize();
+        let state = setup();
+        match handle_query_request(message, state) {
+            Ok(result) => {
+                println!("{:#?}", &result.clone().unwrap());
+                assert!(result.clone().unwrap().body.get("disclosures").is_some());
+                assert!(result
+                    .clone()
+                    .unwrap()
+                    .body
+                    .get("disclosures")
+                    .unwrap()
+                    .is_array());
+                assert!(
+                    result
+                        .clone()
+                        .unwrap()
+                        .body
+                        .get("disclosures")
+                        .unwrap()
+                        .as_array()
+                        .unwrap()
+                        .len()
+                        == 1
+                );
+                let id = serde_json::from_str::<String>(
+                    &result
+                        .unwrap()
+                        .body
+                        .get("disclosures")
+                        .unwrap()
+                        .as_array()
+                        .unwrap()[0]
+                        .as_object()
+                        .unwrap()
+                        .get("id")
+                        .unwrap()
+                        .to_string(),
+                )
+                .unwrap();
+
+                assert_eq!(id, MEDIATION.to_string());
+            }
+            Err(e) => {
+                panic!("This should not occur {:?}", e)
+            }
+        }
+    }
+    #[tokio::test]
+    async fn test_unsupported_feature_type() {
+        let queries = json!({"feature-type": "goal-code", "match": "org.didcomm"});
+        let state = setup();
+        let body = Queries {
+            queries: vec![queries],
+        };
+        let id = Uuid::new_v4().urn().to_string();
+
+        let message = Message::build(id, QUERY_FEATURE.to_string(), json!(body)).finalize();
+
+        match handle_query_request(message, state) {
+            Ok(_) => {
+                panic!("This should'nt occur");
+            }
+            Err(_) => {}
+        }
+    }
+}
```

Comment:
Please include additional test scenarios, such as various message types from the same protocol (to verify that only the protocols are returned and only once), queries that do not match the supported protocols, bad requests, unsupported **feature-type** to see if your code behaves as expected.